pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go

```diff
@@ -204,13 +204,14 @@ var _ = Describe("VirtualMachineInstance Mutator", func() {
 		}
 		_, vmiSpec, _ := getMetaSpecStatusFromAdmit(arch)
 
-		if webhooks.IsPPC64(vmiSpec) {
-			Expect(vmiSpec.Domain.Machine.Type).To(Equal("pseries"))
-		} else if webhooks.IsARM64(vmiSpec) {
+		switch vmiSpec.Architecture {
```

Comment:
This is not directly related to the use of `switch`, but it is odd to add conditions like this here.
The input of `arch` and an expected "Type" should be passed in so the assertion could be done without these conditions. Similar to what was done in the vm-mutator tests.

Reply:
I needed to look through it again, but the reason is because the "Type" is not known for the last test:
```
Entry("when architecture is not specified", "", v1.DefaultCPUModel))
```

Reply:
+1

I suggest to remove the switch and to add another arg to the table's func that contains the expected machine type. Similar tt how it is done with the cpu model already.

Reply:
As i said [here](https://github.com/kubevirt/kubevirt/pull/13852/files#r1959291298), the type is not known for the last testcase. This depends then on the architecture of the host.
To make the type another parameter, i would need to drop the unknown architecture testcase.

Reply:
I have had some time to follow the code that sets the machineType and found a solution.

> Not the prettiest, but the test case below overrides `webhooks.Arch`. Would that be an option? This way you can predict the default architecture.

Unfortunately that value is not used.

What i found that works is, to use the same function that is called as part of defaults to set the machineType.
```
mutator.ClusterConfig.GetMachineType(rt.GOARCH)
```
This at least removes the switch from the test and essentially uses the switch in `virt-config` instead. This way we test it at least calls the function and we make sure it works correctly by testing it with the specific arches first.
From my side, i think this does not change what the test covers so it should be good.